tasks/Task.cpp

```diff
@@ -13,37 +87,528 @@ Task::~Task()
+    host_net_info["push_msg_port"] = host_config.push_msg_port;
+    host_net_info["point_data_ip"] = host_config.point_data_ip;
+    host_net_info["point_data_port"] = host_config.point_data_port;
+    host_net_info["imu_data_ip"] = host_config.imu_data_ip;
+    host_net_info["imu_data_port"] = host_config.imu_data_port;
+    host_net_info["log_data_ip"] = host_config.log_data_ip;
+    host_net_info["log_data_port"] = host_config.log_data_port;
+
+    lidar["lidar_net_info"] = lidar_net_info;
+    lidar["host_net_info"] = host_net_info;
+
+    switch (_lidar_model.get()) {
+        case LidarModel::HAP_T1:
+            data["HAP"] = lidar;
+            break;
+
+        case LidarModel::HAP_TX:
+            data["HAP"] = lidar;
+            break;
+
+        case LidarModel::MID_360:
+            data["MID360"] = lidar;
+            break;
+
+        default:
+            LOG_ERROR_S << "Device Model configuration not found." << endl;
+            return "";
+    }
+
+    while (true) {
+        auto time = base::Time::now();
+        std::string file_name = "livox-" + getName() + "-" +
+                                time.toString(base::Time::Resolution::Milliseconds,
+                                    base::Time::DEFAULT_FORMAT) +
+                                ".json";
+        {
+            std::fstream file_id(file_name.c_str(), ios_base::in);
+            if (file_id) {
+                usleep(10000);
+                continue;
+            }
+        }
+
+        std::fstream file_id(file_name.c_str(), ios_base::out);
+        Json::StyledWriter styledWriter;
+        file_id << styledWriter.write(data);
+        return file_name;
+    }
+    return "";
+}
+
+template <typename F>
+void Task::convertPointcloud(LivoxLidarEthernetPacket const* data,
+    F const* p_point_data,
+    float unit_base)
+{
+    auto now = base::Time::now();
+    if (m_point_cloud.time.isNull()) {
+        m_point_cloud.time = now;
+    }
+
+    for (uint32_t i = 0; i < data->dot_num; i++) {
+        if (!p_point_data[i].x && !p_point_data[i].y && !p_point_data[i].z) {
+            continue;
+        }
+
+        m_point_cloud.points.push_back(base::Point(p_point_data[i].x / unit_base,
+            p_point_data[i].y / unit_base,
+            p_point_data[i].z / unit_base));
+        if (_pointcloud_with_color_based_on_reflexivity.get()) {
+            m_point_cloud.colors.push_back(
+                colorByReflectivity(p_point_data[i].reflectivity));
+        }
+    }
+
+    if (now - m_point_cloud.time >= _integration_time.get()) {
+        _point_cloud.write(m_point_cloud);
+        m_point_cloud.time = now;
+        m_point_cloud.points.clear();
+        m_point_cloud.colors.clear();
+    }
+}
+
+void Task::updatePointcloudReceivedStatus(bool status)
+{
+    {
+        unique_lock<std::mutex> lock(m_pointcloud_sync_mutex);
+        m_pointcloud_received = status;
+    }
+    m_pointcloud_sync_condition.notify_all();
```

Comment:
Even though I think this version does work, I think it would be less suspicious to lock the whole method.